js/ui/endSessionDialog.js

```diff
@@ -417,6 +461,13 @@ var EndSessionDialog = new Lang.Class({
         this._user.disconnect(this._userChangedId);
     },
 
+    _getUpdatesStatus: function()  {
+        if (this._proxy.EntryCount > 0)
+            this._isUpdatesScheduled = true;
+        else
+            this._isUpdatesScheduled = true; /* FIXME: should be false */
+    },
+
```

Comment:
For some reason the `EntryCount` parameter is returning `0` even if there are pending updates scheduled. I don't know why but I can't get past this issue/
As anyone can figure out, I am using `_isUpdatesScheduled` as a flag to know whether there are updates scheduled or not? Is there something wrong with that logic?

Reply:
Capture a D-Bus log of the relevant traffic when `new SchedulerProxy` is called, and see if `GetAll` is called to get the property values from the scheduler. Check there are no error responses in the log. It’s probably easiest to capture the log using [Bustle](https://www.freedesktop.org/wiki/Software/Bustle/).

Reply:
I  figured out what was happening.

I was initially using a single VM for this work and my ongoing work on  gnome-software. More often I used to `killall gnome-software` for `gnome-software`-related work and that used to stop mogwai too (and resetting the properties I think).

Now that if I start mogwai using `service mogwai-scheduled start` and wait for ~1 min, those scheduled downloads objects are spawned (checked using `d-feet`) and `EntryCount` is updated correctly.

Reply:
Was Mogwai crashing, or was it just reaching its inactivity timeout and exiting gracefully?